### PR TITLE
ci: use codecov command line tool

### PR DESCRIPTION
**Pull request by yihau**
#### Summary of Changes

use codecov command line tool to upload coverage results.

close #33107

### CODE REVIEW

**Comment by yihau** [comment]
a weird thing is that https://www.github.com/solana-labs/solana/pull/33131 works fine with codecov bot now 🤔 
but maybe not a big deal cuz this new method can still ship hehe

File: ci/test-coverage.sh [hunk 1]
```diff
@@ -29,10 +29,7 @@ if [[ -z "$CODECOV_TOKEN" ]]; then
   echo "^^^ +++"
   echo CODECOV_TOKEN undefined, codecov.io upload skipped
 else
-  # We normalize CI to `1`; but codecov expects it to be `true` to detect Buildkite...
-  # Unfortunately, codecov.io fails sometimes:
-  #   curl: (7) Failed to connect to codecov.io port 443: Connection timed out
-  CI=true bash <(while ! curl -sS --retry 5 --retry-delay 2 --retry-connrefused --fail https://codecov.io/bash; do sleep 10; done) -Z -X gcov -f target/cov/lcov.info
+  codecov -t "${CODECOV_TOKEN}"
```
Comment (ryoqun) [inline ci/test-coverage.sh:32]: cool. hacks and hacks are finally gone. :)

File: ci/docker-rust-nightly/Dockerfile [hunk 2]
```diff
@@ -10,4 +10,8 @@ RUN set -x \
  && cargo --version \
  && cargo install grcov \
  && rustc +nightly-$date --version \
- && cargo +nightly-$date --version
+ && cargo +nightly-$date --version \
+ # codecov
+ && curl -Os https://uploader.codecov.io/latest/linux/codecov \
+ && chmod +x codecov \
+ && mv codecov /usr/bin
```
Comment (ryoqun) [inline ci/docker-rust-nightly/Dockerfile:17]: i think we can forgo verifying the published sig of the binary, alrought this practice is rightfully advised according codecov doc.

maybe, we can do that bonus rigidity in distance future.. ;) at least, our security isn't degraded from good-old `curl | bash`. ;)

**Comment by ryoqun** [comment]
> a weird thing is that #33131 works fine with codecov bot now 🤔

yeah, that's odd thing i observe as well as i tested at #33107. very strange...

**Review by ryoqun — APPROVED** [review]

**Comment by ryoqun** [comment]
thanks so much! :)

**Pull request merged**